edb/schema/annos.py

```diff
@@ -282,18 +282,23 @@ def _cmd_tree_from_ast(
         astnode: qlast.DDLOperation,
         context: sd.CommandContext
     ) -> CreateAnnotationValue:
+        from . import scalars as s_scalar
 
         assert isinstance(astnode, qlast.CreateAnnotationValue)
         cmd = super()._cmd_tree_from_ast(schema, astnode, context)
         assert isinstance(cmd, CreateAnnotationValue)
         annoname = sn.shortname_from_fullname(cmd.classname)
 
-        value = qlcompiler.evaluate_ast_to_python_val(
+        value, ir = qlcompiler.evaluate_ast_to_python_val_and_ir(
             astnode.value, schema=schema)
 
-        if not isinstance(value, str):
+        if (
+            not isinstance(ir.stype, s_scalar.ScalarType)
```

Comment:
Use `ir.stype.is_scalar()` and then you don't need an import?

Reply:
Actually we don't even need to check that it's a scalar, do we?

Reply:
We don't.